torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -933,21 +934,64 @@ def _run_param_init_fn():
         self._hook_registered = False
 
     def _init_param_exec_order_wrap_policy(self, *args, **kwargs) -> None:
+        auto_wrap_policy = kwargs["auto_wrap_policy"]
```

Comment:
I am not sure what the way around this is, but this requires that we always pass `"auto_wrap_policy"` and `"module"` as keyword arguments (which is fine since we only call `_init_param_exec_order_wrap_policy()` once and internally). I encountered a similar issue recently.

I am not sure if anyone else knows how to approach this.